src/main/java/spireMapOverhaul/zones/beastslair/BeastsLairZone.java

```diff
@@ -0,0 +1,142 @@
+package spireMapOverhaul.zones.beastslair;
+
+import basemod.BaseMod;
+import basemod.abstracts.CustomSavable;
+import com.badlogic.gdx.graphics.Color;
+import com.evacipated.cardcrawl.modthespire.lib.SpirePatch2;
+import com.megacrit.cardcrawl.dungeons.AbstractDungeon;
+import com.megacrit.cardcrawl.helpers.ImageMaster;
+import com.megacrit.cardcrawl.helpers.MonsterHelper;
+import com.megacrit.cardcrawl.map.MapRoomNode;
+import com.megacrit.cardcrawl.monsters.MonsterGroup;
+import com.megacrit.cardcrawl.random.Random;
+import com.megacrit.cardcrawl.rewards.RewardItem;
+import com.megacrit.cardcrawl.rooms.MonsterRoomElite;
+import spireMapOverhaul.abstracts.AbstractZone;
+import spireMapOverhaul.zoneInterfaces.EncounterModifyingZone;
+import spireMapOverhaul.zoneInterfaces.RewardModifyingZone;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static spireMapOverhaul.SpireAnniversary6Mod.makeID;
+
+public class BeastsLairZone extends AbstractZone implements EncounterModifyingZone, RewardModifyingZone {
+    public static final String ID = "BeastsLair";
+    public static final String SaveID = makeID("BeastsLairBosses");
+    public static ArrayList<String> bossList = new ArrayList<>();
+
+    public BeastsLairZone() {
+        super(ID, Icons.MONSTER);
+        this.width = 1;
+        this.maxWidth = 1;
+        this.height = 1;
+        this.maxHeight = 1;
+        iconTexture = ImageMaster.MAP_NODE_ELITE;
+
+    }
+
+    public static void clearBossList() {
+        bossList.clear();
+    }
+
+
+    public static void initializeSaveFields() {
+        BaseMod.addSaveField(SaveID, new CustomSavable<ArrayList<String>>() {
+            @Override
+            public ArrayList<String> onSave() {
+                return BeastsLairZone.bossList;
+            }
+
+            @Override
+            public void onLoad(ArrayList<String> strings) {
+                if (strings != null) {
+                    BeastsLairZone.bossList = strings;
+                }
+            }
+        });
+    }
+
+    @Override
+    protected boolean canIncludeTreasureRow() {
+        return false;
+    }
+
+    @Override
+    protected boolean canIncludeFinalCampfireRow() {
+        return false;
+    }
+
+    @Override
+    protected boolean canIncludeEarlyRows() {
+        return false;
+    }
+
+    @Override
+    public void manualRoomPlacement(Random rng) {
+        for (MapRoomNode node : nodes) {
+            node.setRoom(new MonsterRoomElite());
+        }
+    }
+
+    @Override
+    public MonsterGroup changeEncounter(MonsterGroup monsterGroup) {
```

Comment:
I think there will be a few oddities with this approach that need to be fixed. In particular, anything that checks the room type to determine what to do will think this is an elite fight, even though it really isn't. So for example, Black Star will give you an extra relic, even though you're fighting a boss, not an elite.

I'm sure there's a few different ways you can fix this issue.

One would be to make this an event (like Mind Bloom) rather than a fight. You can use `ForcedEventRoom` in this project (which I made for another zone) to do that. Since that follows a base game pattern, it's a good way to ensure that nothing too weird happens. You'd have to do a bit of work to make a little event for this, but that has some plusses too -- it means you get to write a bit of flavor about entering the Beast's Lair. This would also have the benefit of making this node look a bit different on the map (it will be a ? with little flames flickering behind it).

Reply:
I think that would probably look better on the map too. I'll switch to that.